mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/dependencies/SurefireProviderResolver.java

```diff
@@ -0,0 +1,85 @@
+package org.arquillian.smart.testing.mvn.ext.dependencies;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.maven.model.Dependency;
+import org.arquillian.smart.testing.configuration.Configuration;
+import org.arquillian.smart.testing.logger.Log;
+import org.arquillian.smart.testing.logger.Logger;
+import org.arquillian.smart.testing.known.surefire.providers.KnownProvider;
+
+import static org.arquillian.smart.testing.known.surefire.providers.KnownProvider.JUNIT_4;
+import static org.arquillian.smart.testing.known.surefire.providers.KnownProvider.JUNIT_47;
+import static org.arquillian.smart.testing.known.surefire.providers.KnownProvider.JUNIT_5;
+import static org.arquillian.smart.testing.known.surefire.providers.KnownProvider.TESTNG;
+
+class SurefireProviderResolver {
+
+    private final Logger log = Log.getLogger();
+    private final List<SurefireProviderDefinition> providers = new ArrayList<>();
+
+    SurefireProviderResolver(Configuration configuration) {
+        providers.add(createKnownProviderDependency(JUNIT_4));
+        providers.add(createKnownProviderDependency(JUNIT_47));
+        providers.add(createKnownProviderDependency(JUNIT_5));
+        providers.add(createKnownProviderDependency(TESTNG));
+
+        if (configuration.getCustomProviders().length != 0) {
```

Comment:
Is this `if` necessary? Streams take care of this so it will iterate just 0 times. I know that in terms of performance it is better to have the if thing but I don't know if the gain is so much.

Reply:
zombie - I didn't use the stream in my first version of the implementation :-)